common/types/shared.go

```diff
@@ -2929,6 +2929,8 @@ const (
 	EncodingTypeThriftRW EncodingType = iota
 	// EncodingTypeJSON is an option for EncodingType
 	EncodingTypeJSON
+	// EncodingTypeProto3 is an option for EncodingType
+	EncodingTypeProto3
```

Comment:
doesn't seem to be reflected in the `String()` / `UnmarshalText(...)` implementations above.
generally speaking, it may also be worth adding an 'undefined' zero value, for the same kinds of reason as proto3.  better to detect invalid init immediately rather than e.g. after a database round-trip.